feat: change docker image

patsys opened the pull request:
Concerns #28252

patsys committed:
feat: change docker image

Concerns #28252

christiangebhard commented on Dockerfile:
```diff
@@ -1,8 +1,31 @@
+FROM alpine:3  
+
+LABEL io.k8s.description="helm-wrapper" \ 
+      io.k8s.display-name="heml-wrapper"
```
```suggestion
      io.k8s.display-name="helm-wrapper"
```

christiangebhard commented on Dockerfile:
```diff
@@ -1,8 +1,31 @@
+      io.k8s.display-name="heml-wrapper"
+
+COPY --from=0 /go/src/github.com/opskumu/helm-wrapper/bin/helm-wrapper /helm-wrapper
+COPY --from=0  /go/src/github.com/opskumu/helm-wrapper/config-example.yaml /config.yaml
```
Do we still need the example config file?

miriSch commented on Dockerfile:
```diff
@@ -1,8 +1,31 @@
+  make \
+  musl-dev \
+  go
+#   paxmark bison flex texinfo gawk zip gmp-dev mpfr-dev mpc1-dev zlib-dev libucontext-dev linux-headers isl-dev gcc-gnat-bootstrap
```
remove commented out packages

miriSch commented on Dockerfile:
```diff
@@ -1,8 +1,31 @@
+    PATH=/go/bin:$PATH
+
+RUN  mkdir -p ${GOPATH}/src ${GOPATH}/bin \
+  && git clone https://github.com/opskumu/helm-wrapper $GOPATH/src/github.com/opskumu/helm-wrapper \
```
why do you have to clone the original repository if the current repository is already a fork?

patsys replied:
is from an old design where the repo has not yet been forked, I'll adjust.

miriSch commented on Dockerfile:
```diff
@@ -1,8 +1,31 @@
+RUN  mkdir -p ${GOPATH}/src ${GOPATH}/bin \
+  && git clone https://github.com/opskumu/helm-wrapper $GOPATH/src/github.com/opskumu/helm-wrapper \
+  && cd $GOPATH/src/github.com/opskumu/helm-wrapper \
+  && make build \
+  && make build-linux
```
You don't need both lines here. Since both the intermediate and the resulting docker container are based on an alpine linux, both `make` commands should lead to the same result.

miriSch commented on Makefile:
```diff
@@ -16,7 +16,6 @@ build-windows:
 
 # build docker image
 build-docker:
-	GOOS=linux GOARCH=amd64 go build -ldflags ${LDFLAGS} -o bin/${BINARY_NAME}
```
`build-docker` is not used. Why do you modify it?

patsys replied:
to create the docker image with `make build-docker` and not only with `docker build -t helm-wrapper .`

miriSch replied:
So the command `make build-docker` did not work as expected before? If you fixed this, why did you change the readme?

patsys replied:
For me it is simpler run the docker build command with the  the tag and can change this direct to upload to a registry, other I must make Variable/Parameter or rename after build.

Commit:
chore: Dockerfile fix typo, refectoring

Concerns #28252

miriSch commented on Dockerfile:
```diff
@@ -1,31 +1,27 @@
-  && cd $GOPATH/src/github.com/opskumu/helm-wrapper \
-  && make build \
-  && make build-linux
+COPY .  $GOPATH/src/github.com/opskumu/helm-wrapper 
+WORKDIR $GOPATH/src/github.com/opskumu/helm-wrapper
```
I think the path could be shorter than `$GOPATH/src/github.com/opskumu/helm-wrapper` and it should be easy to recognize when used in this line: `COPY --from=0 /go/src/github.com/opskumu/helm-wrapper/bin/helm-wrapper /helm-wrapper`

patsys committed:
feat: add ca and change user

#Concerns 28252

christiangebhard commented on Dockerfile:
```diff
@@ -1,8 +1,46 @@
-FROM centos:7
+OM alpine:3 as build
```
```suggestion
FROM alpine:3 as build
```

miriSch committed:
feat: Dockerfile fixes

fix typo
add home directory for user (otherwise helm wrapper returns error permission denied)

Concerns #28252

miriSch committed:
WIP: try refactor dockerfile

miriSch committed:
fixed file missing for build

miriSch committed:
removed labels

miriSch committed:
remove certificate configuration (should be done from helm chart) and extra user

miriSch committed:
use release mode instead of debug mode

miriSch committed:
combine COPY commands to produce less layers
